app/models/user.rb

```diff
@@ -2,4 +2,16 @@ class User < ApplicationRecord
   has_many :likes, foreign_key: 'author_id', dependent: :destroy
   has_many :posts, foreign_key: 'author_id', dependent: :destroy
   has_many :comments, foreign_key: 'author_id', dependent: :destroy
+
+  after_create :update_posts_counter
+  after_destroy :update_posts_counter
+
+  def update_posts_counter
+    self.posts_counter = posts.count
+    save
+  end
```

Comment:
- [OPTIONAL] As highlighted by the previous reviewer, you don't need these methods because they will be done by the methods in the Post model. I would advise you not to repeat yourself. 👍 💻